File: OCaml Basics/23809.ml
open Scanf
open Printf
;;
let n = read_int()
;;
let f n s =
    for j=1 to n do
        printf s
    done
;;

let s1 n =
    for i=1 to n do
        f n "@";
        f (3*n) " ";
        f n "@";
        f 1 "\n"
    done
;;

let s2 n =
    for i=1 to n do
        f n "@";
        f (2*n) " ";
        f n "@";
        f 1 "\n"
    done
;;

let s3 n =
    for i=1 to n do
        f (3*n) "@";
        f 1 "\n"
    done
;;

s1 n;s2 n;s3 n;s2 n;s1 n
;;

File: C Basics/23809.c
#include <stdio.h>;

void f(int n,int s){
    for(int i=0;i<n;i++){
        printf("%c", s);
    }
}

void s1(int n){
    for(int i=0;i<n;i++){
        f(n, 64);
        f(3*n, 32);
        f(n, 64);
        f(1, 10);
    }
}

void s2(int n){
    for(int i=0;i<n;i++){
        f(n, 64);
        f(2*n, 32);
        f(n, 64);
        f(1, 10);
    }
}

void s3(int n){
    for(int i=0;i<n;i++){
        f(3*n, 64);
        f(1, 10);
    }
}

int main(){
    int n;
    scanf("%d",&n);
    s1(n);s2(n);s3(n);s2(n);s1(n);
    return 0;
}
